src/api/providers/openai.ts

```diff
@@ -212,12 +238,23 @@ export class OpenAiHandler extends BaseProvider implements SingleCompletionHandl
 
 	async completePrompt(prompt: string): Promise<string> {
 		try {
+			// Add Azure AI Inference check within this method
+			let urlHost: string
+			try {
+				urlHost = new URL(this.options.openAiBaseUrl ?? "").host
+			} catch (error) {
+				urlHost = ""
+			}
+			const isAzureAiInference = urlHost.endsWith(".services.ai.azure.com")
```

Comment:
Think we could make a helper method for this logic so we didn't have to repeat it so much?